File: config/config-sbc2ha-new/src/main/java/com/dfi/sbc2ha/config/sbc2ha/definition/action/OutputActionConfig.java
package com.dfi.sbc2ha.config.sbc2ha.definition.action;

import com.dfi.sbc2ha.config.sbc2ha.definition.enums.ActionOutputType;
import com.dfi.sbc2ha.config.sbc2ha.definition.enums.ActionType;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;
import lombok.EqualsAndHashCode;

@EqualsAndHashCode(callSuper = true)
@Data
public class OutputActionConfig extends ActionConfig {

    @JsonProperty("output")
    int output;

    @JsonProperty("action_output")
    ActionOutputType actionOutput;

    public OutputActionConfig() {
        super();
        action = ActionType.OUTPUT;
    }
}

File: config/editor/src/app/definition/action/OutputActionConfig.ts
import {ActionConfig} from "./ActionConfig";
import {ActionOutputType} from "../enums/ActionOutputType";
import {ActionType} from "../enums/ActionType";
import {Expose} from "class-transformer";

export class OutputActionConfig extends ActionConfig {
    output: number;
    @Expose({name: "action_output"})
    actionOutput: ActionOutputType;


    constructor(output: number, actionOutput: ActionOutputType) {
        super(ActionType.OUTPUT);
        this.output = output;
        this.actionOutput = actionOutput;
    }
}
